prusti-viper/src/verifier.rs

```diff
@@ -228,148 +231,28 @@ impl<'v, 'r, 'a, 'tcx> Verifier<'v, 'r, 'a, 'tcx> {
-            let source_filename = source_path.file_name().unwrap().to_str().unwrap();
-            let mut dump_path = PathBuf::from("viper_program");
-            let num_parents = config::num_parents_for_dumps();
-            if num_parents > 0 {
-                // Take `num_parents` parent folders and add them to `dump_path`
-                let mut components = vec![];
-                if let Some(abs_parent_path) = canonicalize(&source_path)
-                    .ok()
-                    .and_then(|full_path| full_path.parent().map(|parent| parent.to_path_buf()))
-                {
-                    components.extend(
-                        abs_parent_path
-                            .ancestors()
-                            .flat_map(|path| path.file_name())
-                            .take(num_parents as usize)
-                            .map(|x| x.to_os_string())
-                            .collect::<Vec<_>>()
-                            .into_iter()
-                            .rev(),
-                    );
-                } else {
-                    components.push(OsString::from("io_error"))
-                }
-                for component in components {
-                    dump_path.push(component);
-                }
-            }
-            info!("Dumping Viper program to the {:?} folder", dump_path);
-            log::report(
-                dump_path.to_str().unwrap(),
-                format!("{}.vpr", source_filename),
-                self.ast_utils.pretty_print(program),
-            );
+        for &proc_id in task.procedures.iter().rev() {
+            self.encoder.queue_procedure_encoding(proc_id);
         }
+        self.encoder.process_encoding_queue();
 
-        let duration = start.elapsed();
-        info!(
-            "Construction of JVM objects successful ({}.{} seconds)",
-            duration.as_secs(),
-            duration.subsec_millis() / 10
-        );
-        let start = Instant::now();
+        stopwatch.finish();
 
-        let verification_result: viper::VerificationResult = self.verifier.verify(program);
+        let encoding_errors_count = self.encoder.count_encoding_errors();
 
-        let duration = start.elapsed();
-        info!(
-            "Verification complete ({}.{} seconds)",
-            duration.as_secs(),
-            duration.subsec_millis() / 10
-        );
+        let mut program = self.encoder.get_viper_program();
+        if config::simplify_encoding() {
+            program = program.optimized();
+        }
+        let verification_result = verify(program);
 
         let verification_errors = match verification_result {
             viper::VerificationResult::Failure(errors) => errors,
-            _ => vec![],
+            viper::VerificationResult::ConsistencyErrors(errors) => {
+                errors.iter().for_each(|e| error!("{}", e));
+                panic!("Consistency errors. The encoded Viper program is incorrect.");
```

Comment:
Instead of `panic`, could you try using `PrustiError::internal("...", DUMMY_SP).emit();`, returning `Failure`?